packages/messaging/README.md

```diff
@@ -127,10 +136,246 @@ Handling push notifications must be done in native (Android/iOS) code for the Re
+
+## In App Messages API reference
+### refreshInAppMessages
+Initiates a network call to retrieve remote In-App Message definitions.
+
+**Syntax**
+```javascript
+refreshInAppMessages()
+```
+
+**Example**
+```javascript
+Messaging.refreshInAppMessages();
+```
+
+### setMessagingDelegate
+Sets the UI Message delegate to listen the Message lifecycle events.
+
+**Syntax**
+```javascript
+setMessagingDelegate(delegate: MessagingDelegate);
+```
+
+**Example**
+```javascript
+const messagingDelegate = {
+
+    onShow(message: Message) {
+       // Action after message is displayed.
+    },
+
+    onDismiss(message: Message) {
+       // Action after message is dismissed.
+    },
+
+    shouldShowMessage(message: Message) {                  
+        return true; //Return true to show the Message else return false
+    },
+
+    urlLoaded(url: string, message: Message) {
+       // Action after message loads an URL
+    }
+  };
+        
+  Messaging.setMessagingDelegate(messagingDelegate);
+```
+
+### saveMessage
+Call to the saveMessage leads to in-memory caching of the Message object. Cached Message object can be later use to show message. This function should be called from **shouldShowMessage** of MessagingDelegate.
+
+**Syntax**
+```javascript
+saveMessage(message: Message);
+```
+
+**Example**
+```javascript
+Messaging.saveMessage(message);
+```
+
+## Handling In App Messages using Message Object
+The Message Object passed to the Messaging delegate contains the following functions to handle a message
+
+### show
+Signals to the UIServices that the message should be shown.
+
+**Syntax**
+```javascript
+show();
+```
+
+**Example**
+```javascript
+var message: Message;
+message.show();
+```
+
+### dismiss
+Signals to the UIServices that the message should be dismissed.
+
+**Syntax**
+```javascript
+dismiss(suppressAutoTrack: ?boolean = false);
+```
+
+**Example**
+```javascript
+var message: Message;
+message.dismiss(true);
+```
+
+### track
+Generates an Edge Event for the provided interaction and eventType.
+
+**Syntax**
+```javascript
+track(interaction: ?string, eventType: MessagingEdgeEventType);
+```
+
+**Example**
+```javascript
+var message: Message;
+message.track("sample text", MessagingEdgeEventType.IN_APP_DISMISS);
+```
+
+### handleJavascriptMessage
+Adds a handler for Javascript messages sent from the message's webview.
+
+**Syntax**
+```javascript
+handleJavascriptMessage(name: string) : Promise<?any>;
+```
+
+**Example**
+```javascript
+var message: Message;
+message.handleJavascriptMessage("test").then((data) => {
+
+});
+```
+
+### setAutoTrack
+Enables/Disables the autotracking for the Message events.
+
+**Syntax**
+```javascript
+setAutoTrack(autoTrack: boolean);
+```
+
+**Example**
+```javascript
+var message: Message;
+message.setAutoTrack(true);
+```
+
+### clearMessage
```

Comment:
wondering if maybe we should change the name of this method to just `clear`?  

`message.clearMessage();` sounds a little redundant

`message.clear();` sounds better

🤷

Reply:
It does sound better.